src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -57,9 +64,21 @@ static PSVersionInfo()
         {
             s_psVersionTable = new PSVersionHashTable(StringComparer.OrdinalIgnoreCase);
 
+            // Get '6.0.0-beta.7 Commits: 29 SHA: 52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c'.
```

Comment:
The format could also be `6.0.0-beta.7 SHA: 52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c` without the Commit count. It's better to mention that too.
How about:
```
// Get ProductVersion of the assembly. The product version string can be one of the following format examples:
//    when powershell is built from a commit -- '6.0.0-beta.7 Commits: 29 SHA: 52c6b...'
//    when powershell is built from a release tag -- '6.0.0-beta.7 SHA: f1ec9...'
```

Reply:
Done.